pkg/provider/provider.go

```diff
@@ -308,37 +331,67 @@ func (p *Provider) GetSecretsStoreObjectContent(ctx context.Context, attrib, sec
 			return nil, err
 		}
 
-		// fetch the object from Key Vault
-		content, newObjectVersion, err := p.getKeyVaultObjectContent(ctx, kvClient, keyVaultObject, *vaultURL)
-		if err != nil {
-			return nil, err
-		}
+		// if version history isn't greater than 1, maintain the current behavior
+		if keyVaultObject.ObjectVersionHistory <= 1 {
+			file, err := p.getSecretFile(ctx, kvClient, filePermission, baseFileName, keyVaultObject, keyVaultObject.ObjectVersion, *vaultURL)
 
-		objectContent, err := getContentBytes(content, keyVaultObject.ObjectType, keyVaultObject.ObjectEncoding)
-		if err != nil {
-			return nil, err
-		}
+			if err != nil {
+				return nil, err
+			}
 
-		// objectUID is a unique identifier in the format <object type>/<object name>
-		// This is the object id the user sees in the SecretProviderClassPodStatus
-		objectUID := getObjectUID(keyVaultObject.ObjectName, keyVaultObject.ObjectType)
+			// these files will be returned to the CSI driver as part of gRPC response
+			files = append(files, file)
+			klog.V(5).InfoS("added file to the gRPC response", "file", baseFileName, "pod", klog.ObjectRef{Namespace: podNamespace, Name: podName})
+		} else {
+			// otherwise we'll generate multiple files for this object
+			secretVersions, err := p.getKeyVaultSecretVersions(ctx, kvClient, keyVaultObject, *vaultURL)
+			if err != nil {
+				return nil, err
+			}
 
-		// these files will be returned to the CSI driver as part of gRPC response
-		files = append(files, SecretFile{
-			Path:     fileName,
-			Content:  objectContent,
-			FileMode: filePermission,
-			UID:      objectUID,
-			Version:  newObjectVersion,
-		})
-		klog.V(5).InfoS("added file to the gRPC response", "file", fileName, "pod", klog.ObjectRef{Namespace: podNamespace, Name: podName})
+			for i, version := range secretVersions {
+				fileName := fmt.Sprintf("%s/%d", baseFileName, i)
```

Comment:
This is different from what the PR description says:
```
It will then put each secret into a file named {objectName\objectAlias}/{secretVersion}
```

Here you're using the index as suffix for the file name instead?